### PR TITLE
feat(app): add visual run progress meter replace run log with preview

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Remove run log in favor of new performant run preview tab, which presents robot-side analysis
commands. Add new run progress meter with ticks that represent manual interventions in the course of
the protocol.

re [RLAB-241](https://opentrons.atlassian.net/browse/RLAB-241)

https://user-images.githubusercontent.com/4731953/216712840-d7c3d3dc-df01-4bbd-995b-8c0f901717b4.mov

https://user-images.githubusercontent.com/4731953/216712861-da052491-edff-4cf4-900e-1db48e92cbc3.mov

# Changelog

- Implement the Run Progress Meter with intervention ticks that now appears at the bottom of the Protocol Run Header
- Replace "Run Log" tab and route with "Run Preview" tab and route
- Remove all Run Log components and children
- Include new package [react-viewport-list](https://www.npmjs.com/package/react-viewport-list) to handle virtualization of run preview commands and seamless "jump to index" functionality
- Consolidate and pare down PE command -> human-readable description with new  `CommandText` component that should handle any and all `commandType`'s

# Review requests

- View the run details page for unstarted, started, paused, canceled, and completed runs and confirm that the run preview and progress bar behave as outlined in the design spec.

# Risk assessment
High, this is a large inclusion and large removal within the app. 

[RLAB-241]: https://opentrons.atlassian.net/browse/RLAB-241?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/atoms/buttons/index.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 export { PrimaryButton } from './PrimaryButton'
 export { SecondaryButton } from './SecondaryButton'
 export { TertiaryButton } from './TertiaryButton'
-export { SecondaryTertiaryButton } from './SecondaryTertiaryButton'
+export { QuaternaryButton } from './QuaternaryButton'
```
Comment (b-cooper) [inline app/src/atoms/buttons/index.ts:4]: This renaming is for temporary clarity, though the plan is to restructure the tertiary and quaternary button atoms to match their function in the design system, as simply variants of the primary and secondary buttons.  In addition to variant props like `isDangerous` there would be a new prop for `rounded`/`pillShaped` .

File: app/src/atoms/buttons/SecondaryButton.tsx [hunk 2]
```diff
@@ -1,24 +1,37 @@
 import styled from 'styled-components'
```
Comment (b-cooper) [inline app/src/atoms/buttons/SecondaryButton.tsx:1]: This refactor decouples the app's `SecondaryButton` component from the chain of legacy components contained in the components library (e.g. `Btn`, `SecondaryBtn`, `NewSecondaryBtn`). All of the functionality and appearance should be identical except for the addition of the `isDangerous` prop which should replace the need for the `AlertSecondaryButton`

File: app/src/organisms/CommandText/MoveLabwareCommandText.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+interface MoveLabwareCommandTextProps {
+  command: MoveLabwareRunTimeCommand
+  robotSideAnalysis: CompletedProtocolAnalysis
+}
+export function MoveLabwareCommandText(
+  props: MoveLabwareCommandTextProps
+): JSX.Element {
+  const { t } = useTranslation('protocol_command_text')
+  const { command, robotSideAnalysis } = props
+  const { labwareId, newLocation, strategy } = command.params
```
Comment (shlokamin) [inline app/src/organisms/CommandText/MoveLabwareCommandText.tsx:19]: ooo a `strategy` param! how cool

File: app/src/organisms/CommandText/PipettingCommandText.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+import type {
+  CompletedProtocolAnalysis,
+  AspirateRunTimeCommand,
+  DispenseRunTimeCommand,
+  BlowoutRunTimeCommand,
+  MoveToWellRunTimeCommand,
+  DropTipRunTimeCommand,
+  PickUpTipRunTimeCommand,
+} from '@opentrons/shared-data'
+import { getLoadedLabware } from './utils/accessors'
```
Comment (shlokamin) [inline app/src/organisms/CommandText/PipettingCommandText.tsx]: nit: this should go before the type imports. also do we not want this exported at the top level of the `utils` dir like `getLabwareName` and `getLabwareDisplayLocation`?

File: app/src/organisms/CommandText/index.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+import { StyledText } from '../../atoms/text'
+import { LoadCommandText } from './LoadCommandText'
+import { PipettingCommandText } from './PipettingCommandText'
+import { TemperatureCommandText } from './TemperatureCommandText'
+import { MoveLabwareCommandText } from './MoveLabwareCommandText'
+
+import type { RunTimeCommand } from '@opentrons/shared-data'
+import type { CompletedProtocolAnalysis } from '@opentrons/shared-data/js'
+
+const SIMPLE_T_KEY_BY_COMMAND_TYPE: {
```
Comment (shlokamin) [inline app/src/organisms/CommandText/index.tsx]: nit: i can was confused what the "T" was for when i initially read this. maybe rename to `SIMPLE_TRANSLATION_KEY_BY_COMMAND_TYPE`?

File: app/src/organisms/RunProgressMeter/Tick.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    index,
+    count,
+    range,
+    firstCommandType,
+    makeHandleJumpToStep,
+    total,
+  } = props
+  const { t } = useTranslation('run_details')
+
+  const tKeyByCommandType: {
```
Comment (shlokamin) [inline app/src/organisms/RunProgressMeter/Tick.tsx]: nit: same thing here "t", also should this variable be capiralized?

**Review by shlokamin — APPROVED** [review]
wowza! this was a lot to read through but these changes leave the run page in a much better state than it was in before. thanks for doing this! 

for posterity: the current run log is broken and RSS is going to smoke test these changes in edge (when this merges) in the coming days

File: app/src/organisms/CommandText/PipettingCommandText.tsx (hunk 4 above)
Comment (b-cooper) [reply app/src/organisms/CommandText/PipettingCommandText.tsx]: I was aiming to keep them separate as the top level `utils` provides simple `string | null` values to be used directly in the view layer.  I wanted to sequester the functions that deal with the actual shape of the analysis.  Technically speaking I'd probably want these `accessors` to only be imported within the `utils` dir itself.  We will have to revisit this one soon when we have to support dynamic display location for `move_labware` commands etc.

**Pull request merged**